test/decl/overload.swift

```diff
@@ -494,135 +498,129 @@ extension P3 {
   subscript(i: Int) -> String { return "" }
 }
 
-struct SR7250<T> : P3 {}
+struct S_49798<T> : P3 {}
 
-extension SR7250 where T : P3 {
-  var i: Int { return 0 }
-  subscript(i: Int) -> String { return "" }
+extension S_49798 where T : P3 {
+  var i: Int { get {} }
+  subscript(i: Int) -> String { get {} }
 }
 
-// SR-10084
+// https://github.com/apple/swift/issues/52486
 
-struct SR_10084_S {
+struct S_52486 {
   let name: String
 }
 
-enum SR_10084_E {
-  case foo(SR_10084_S) // expected-note {{'foo' previously declared here}}
+enum E_52486 {
+  case foo(S_52486) // expected-note {{'foo' previously declared here}}
     
-  static func foo(_ name: String) -> SR_10084_E { // Okay
-    return .foo(SR_10084_S(name: name))
+  static func foo(_ name: String) -> E_52486 { // Okay
+    return .foo(S_52486(name: name))
   }
 
-  func foo(_ name: Bool) -> SR_10084_E { // Okay
-    return .foo(SR_10084_S(name: "Test"))
+  func foo(_ name: Bool) -> E_52486 { // Okay
+    return .foo(S_52486(name: "Test"))
   }
 
-  static func foo(_ value: SR_10084_S) -> SR_10084_E { // expected-error {{invalid redeclaration of 'foo'}}
+  static func foo(_ value: S_52486) -> E_52486 { // expected-error {{invalid redeclaration of 'foo'}}
     return .foo(value)
   }
 }
 
-enum SR_10084_E_1 {
-  static func foo(_ name: String) -> SR_10084_E_1 { // Okay
-    return .foo(SR_10084_S(name: name))
+enum E1_52486 {
+  static func foo(_ name: String) -> E1_52486 { // Okay
+    return .foo(S_52486(name: name))
   }
 
-  static func foo(_ value: SR_10084_S) -> SR_10084_E_1 { // expected-note {{'foo' previously declared here}}
+  static func foo(_ value: S_52486) -> E1_52486 { // expected-note {{'foo' previously declared here}}
     return .foo(value)
   }
 
-  case foo(SR_10084_S) // expected-error {{invalid redeclaration of 'foo'}}
+  case foo(S_52486) // expected-error {{invalid redeclaration of 'foo'}}
 }
 
-enum SR_10084_E_2 {
+enum E2_52486 {
   case fn(() -> Void) // expected-note {{'fn' previously declared here}}
 
-  static func fn(_ x: @escaping () -> Void) -> SR_10084_E_2 { // expected-error {{invalid redeclaration of 'fn'}}
-    fatalError()
-  }
+  static func fn(_ x: @escaping () -> Void) -> E2_52486 {} // expected-error {{invalid redeclaration of 'fn'}}
 
-  static func fn(_ x: @escaping () -> Int) -> SR_10084_E_2 { // Okay
-    fatalError()
-  }
+  static func fn(_ x: @escaping () -> Int) -> E2_52486 {} // Okay
```

Comment:
I am taking the liberty to remove some of these dummy returns within code I modified, since they present unnecessary noise and, conveniently, are mostly optional until after semantic analysis. Let me know If you consider this a bit too invasive.